sycl/include/CL/sycl/exception.hpp

```diff
@@ -12,40 +12,45 @@
 
 #include <CL/sycl/detail/common.hpp>
 #include <CL/sycl/stl.hpp>
-#include <exception>
 
 namespace cl {
 namespace sycl {
 
+// Forward declaration
 class context;
 
 struct exception {
   exception() = default;
 
-  const char *what() const noexcept { return MMsg.c_str(); }
+  const char *what() const noexcept;
```

Comment:
Why you want to move this function to the library?

Reply:
Because other three functions in the library.
So why not?